master/buildbot/schedulers/trysched.py

```diff
@@ -117,6 +120,9 @@ def activate(self):
 
     @defer.inlineCallbacks
     def deactivate(self):
+        if not self.enabled:
```

Comment:
I think waiting for the deactivate as you mention below should fix this. I just don't want the enabled flag to be out of sync with reality.